### DISCUSSION TITLE
Record Live Broadcasts: Duration limit ?

### DISCUSSION THREAD

**Discussion by vitorfernandes1976** [AzuraCast/AzuraCast#5166] [Q&A]
### Installation Method

Docker Installation

### AzuraCast Release Channel

Stable Channel

### Current AzuraCast Version

v0.15.2 Stable

### What happened?

I have a problem (which could be a bug).
I have 15 radio stations created, of which 4 radio stations go "live" 24 hours a day.

I have the recording option enabled only at 32kbps (only for legal purposes).

When I download the recordings, I notice that they are never more than 12 hours long...

Are there any limitations on the recording duration?

Thanks.

**Comment by Vaalyn** [comment]
As far as I'm aware there is no strict time limitation.

It could be that the live source buffer in Liquidsoap is running dry after 12 hours (this can happen over time due to small network hiccups) which will lead to your live dj software automatically reconnecting. This would create a new recording since the "dj" disconnects and then reconnects.

You can check the Liquidsoap Logs to see if the DJ software happens to be doing this around the time the recordings stop and start again.

**Reply by vitorfernandes1976** [reply]
In the future I leave a suggestion: Option for legal recording 24 hours a day (professional and legalized radios are required to have recording 24x7). Thank you very much.

**Comment by LordHelmchen666** [comment]
If you want to record 24 hours a day  i think its better to use Liquidsoap itself, because it is more reliable. I do this myself using  this code:

```
# Save Stream, make new file every hour
output.file(
%fdkaac(channels=2, samplerate=44100, vbr=5, afterburner=true, aot="mpeg4_he_aac_v2", transmux="adts", sbr_mode=false),
 "/var/azuracast/stations/mystation/recordings/%Y/%m/%d/%Y-%m-%d-T%H_%M_%S.mp4", 
reopen_when={0m0s}, 
radio)
```

or you can use MP3 also like this:

```
output.file(
%mp3(bitrate=192),
 "/var/azuracast/stations/mystation/recordings/%Y/%m/%d/%Y-%m-%d-T%H_%M_%S.mp3", 
reopen_when={0m0s}, 
radio)

```
Add it at the end of Liquidsoap Configuration. It will create Folders Year/Month/Day with 24 files

**Reply by vitorfernandes1976** [reply]
In the future I leave a suggestion: Option for legal recording 24 hours a day (professional and legalized radios are required to have recording 24x7). I have a dedicated server with 2TB SSD 32 cores 1GB dedicated and a very well configured AzuraCast. Everything perfect. Congratulations. I will be making a donation soon. Thank you very much.

**Comment by LordHelmchen666** [comment]
If someone else needs to record 24/7 and want to auto delete after X days here is my code:
```
# Save Stream, open new file every hour
output.file(
%fdkaac(channels=2, samplerate=44100, vbr=5, afterburner=true, aot="mpeg4_he_aac_v2", transmux="adts", sbr_mode=false),
 "/var/azuracast/stations/YOURSTATION/recordings/archive/%Y-%m-%d/%Y-%m-%d-T%H_%M_%S.mp4", 
reopen_when={0m0s}, 
radio)
# Delete files in /recordings/archive older than 120 days
thread.when(every=3600., changed=false, { true },
    fun () -> list.iter(fun(msg) ->  log(msg, label="archive_cleaner"), 
       list.append(
            process.read.lines("find /var/azuracast/stations/YOURSTATION/recordings/archive/* -type f -mtime +121 -delete"),
            process.read.lines("find /var/azuracast/stations/YOURSTATION/recordings/archive/* -type d -empty -delete")
        )
    )
)
```

**Reply by ItsMitchh** [reply]
Thank you so much for this handy script, I'll add this into our docs in the Liquidsoap area.

**Reply by LordHelmchen666** [reply]
This is taken from https://www.liquidsoap.info/doc-2.0.3/cookbook.html and just slightly modified. So i did not much. ;-)

**Comment by streulma** [comment]
I want to create an output every 6 hours.
So when the recording is longer then 6 hours and a live stream is 12 hours.
So split in 2.
All other recordings from 1 hour or 2 hour should remain normal.

What can I use with reopen_when= ?

output.file(
%mp3(bitrate=320),
 "/var/azuracast/stations/mystation/recordings/%Y-%m-%d-T%H_%M_%S.mp3", 
reopen_when={0m0s}, 
radio)

**Reply by vitorfernandes1976** [reply]
Hello everybody;

Thank you for your help.

After the last update this "update" stopped working.

Can you please help to resolve it?

We are a professional radio group and are required by law to have legal recording 24 hours a day.

Thanks.

Vitor Fernandes

**Reply by LordHelmchen666** [reply]
There have been some changes in LS. I use this Code now

```
# Save Stream, open new file every hour
recording_archive_path = "/var/azuracast/stations/eisradio/recordings/archive"
output.file(
%fdkaac(channels=2, samplerate=44100, bitrate=96, afterburner=false, aot="mpeg4_he_aac_v2", sbr_mode=true),
 {time.string("#{recording_archive_path}/%Y-%m-%d/%Y-%m-%d-T%H_%M_%S.aac")}, radio, reopen_when={0m})
# Delete files in /recordings/archive older than 120 days and delete empty directorys
thread.when(every=3600., changed=false, { true },
    fun () -> list.iter(fun(msg) ->  log(msg, label="archive_cleaner"), 
       list.append(
            process.read.lines("find #{recording_archive_path}/* -type f -mtime +121 -delete"),
            process.read.lines("find #{recording_archive_path}/* -type d -empty -delete")
        )
    )
)

```
Dont forget to change `record_archive_path` to fit your Station!

**Reply by LordHelmchen666** [reply]
@streulma I suggest you to use mp3wrap https://mp3wrap.sourceforge.net/ to merge mp3 files. Or you have to define something  like "if djlive then reopen=xtime else reopen=ytime" in your LS code.

**Comment by vitorfernandes1976** [comment]
Hello, I am very grateful for your help.
I think it's working. Thank you very much.
How do I record MP3 files?

Thanks.

LordHelmchen666 ***@***.***> escreveu no dia sexta,
15/09/2023 à(s) 10:49:

**Comment by LordHelmchen666** [comment]
```
# Save Stream, open new file every hour
recording_archive_path = "/var/azuracast/stations/YOURSTATION/recordings/archive"
output.file(
%mp3(bitrate=128),
 {time.string("#{recording_archive_path}/%Y-%m-%d/%Y-%m-%d-T%H_%M_%S.mp3")}, radio, reopen_when={0m})
# Delete files in /recordings/archive older than 120 days and delete empty directorys
thread.when(every=3600., changed=false, { true },
    fun () -> list.iter(fun(msg) ->  log(msg, label="archive_cleaner"), 
       list.append(
            process.read.lines("find #{recording_archive_path}/* -type f -mtime +121 -delete"),
            process.read.lines("find #{recording_archive_path}/* -type d -empty -delete")
        )
    )
)
```
